keyring/backends/macOS/__init__.py

```diff
@@ -32,36 +33,54 @@ def priority(cls):
             raise RuntimeError("Security API unavailable")
         return 5
 
-    def set_password(self, service, username, password):
+    def set_password(self, service, username, password, keychain=""):
         if username is None:
             username = ''
 
+        if keychain != "":
+            if not exists(keychain):
+                raise KeyringLocked("Can't open specified keychain")
```

Comment:
I'd not use KeyringLocked here... because the keyring isn't locked. You'll want to create a new exception or find one that's appropriate for the failure.